sympy/ntheory/factor_.py

```diff
@@ -963,8 +963,7 @@ def _check_termination(factors, n, limitp1, use_trial, use_rho, use_pm1,
         for b, e in facs.items():
             if verbose:
                 print(factor_msg % (b, e))
-            # int() can be removed when https://github.com/flintlib/python-flint/issues/92 is resolved
-            factors[b] = int(exp*e)
+            factors[b] = exp*e
```

Comment:
> If this means that `pow(int, int, fmpz)` would be used again then it would fail under PyPy with python-flint installed.

To be clear it would raise TypeError rather than returning an incorrect result.

Reply:
> If this means that `pow(int, int, fmpz)` would be used again then it would fail under PyPy with python-flint installed.

I see. Then I don't think we should remove the cast to int. I will just remove the comment.
I don't think `finitefield.py` needs the logic since `gmpy.py` controls the import.

> As an aside python-flint provides integer factorisation so in the case of python-flint would it be better to use that?

The current name `external.gmpy` could be misleading if the flint functions are to be included. Also, the relationship with `gmpy` needs to be sorted out.
sympy has a `factorint`, and I think the problem of how to make it consistent with that will arise.

Reply:
> > As an aside python-flint provides integer factorisation so in the case of python-flint would it be better to use that?
> 
> The current name `external.gmpy` could be misleading if the flint functions are to be included.

It is already misleading. Perhaps it should be changed. I think that something like `sympy.base.ntheory` makes sense.

When originally creating `sympy.external.gmpy` the main intention was just to cover over the difference between the ground types for when gmpy2 is being used or not. I didn't really expect that so much more of ntheory would be replaced by gmpy2. Possibly there should be a `sympy.base.ntheory` module with submodules like `python`, `gmpy` and `flint` for this.

Some other things that flint is used for like polynomials and matrices are too big to be moved to either external or base so that approach is not generally applicable. Perhaps that also applies to a lot of the ntheory functions that have been moved.

It probably also makes sense to move `modularinteger` to sit alongside `pythonmpq` and have a something analogous to `MPZ` and `MPQ` for modular integers.

Reply:
It is not clear what role `sympy.extarnal`, `sympy.base`, and `sympy.ntheory` perform respectively.

Reply:
Thank you for the explanation.
I understand the concept, but I think it will be difficult to achieve.